src/github.com/matrix-org/dendrite/appservice/consumers/roomserver.go

```diff
@@ -142,3 +156,57 @@ func (s *OutputRoomEventConsumer) lookupStateEvents(
 
 	return result, nil
 }
+
+// filterRoomserverEvents takes in events and decides whether any of them need
+// to be passed on to an external application service. It does this by checking
+// each namespace of each registered application service, and if there is a
+// match, adds the event to the queue for events to be sent to a particular
+// application service.
+func (s *OutputRoomEventConsumer) filterRoomserverEvents(events []gomatrixserverlib.Event) error {
+	for _, event := range events {
+		for _, appservice := range appServices {
+			// Check if this event is interesting to this application service
+			if s.appserviceIsInterestedInEvent(event, appservice) {
+				// TODO: Queue this event to be sent off to the application service
+				fmt.Println(appservice.ID, "was interested in", event.Sender(), event.Type(), event.RoomID())
+			}
+		}
+	}
+
+	return nil
+}
+
+// appserviceIsInterestedInEvent returns a boolean depending on whether a given
+// event falls within one of a given application service's namespaces.
+func (s *OutputRoomEventConsumer) appserviceIsInterestedInEvent(event gomatrixserverlib.Event, appservice config.ApplicationService) bool {
+	// Check sender of the event
+	for _, userNamespace := range appservice.NamespaceMap["users"] {
+		if userNamespace.RegexpObject.MatchString(event.Sender()) {
+			return true
+		}
+	}
+
+	// Check room id of the event
+	for _, roomNamespace := range appservice.NamespaceMap["rooms"] {
+		if roomNamespace.RegexpObject.MatchString(event.RoomID()) {
+			return true
+		}
+	}
+
+	// Check all known room aliases of the room the event came from
+	queryReq := api.GetAliasesForRoomIDRequest{RoomID: event.RoomID()}
+	var queryRes api.GetAliasesForRoomIDResponse
+	if err := s.alias.GetAliasesForRoomID(context.TODO(), &queryReq, &queryRes); err == nil {
```

Comment:
You should be threading the context through. `context.TODO()` should only be used when you otherwise don't know quite where to get one yet, but we already have made a context for `UpdateMemberships`, so lets use that one.

Reply:
Gotcha. Still figuring out how context works :)